### PR TITLE
[TwigBridge] fix trimmed spaces before id attributes (were trimmed before)

### PR DESCRIPTION
**Pull request by j3ll3yfi5h**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| License       | MIT

### CODE REVIEW

**Comment by fabpot** [comment]
3.4 is not maintained anymore. Does the same issue exist in 4.4 (which is the oldest maintained Symfonyversion)?

**Comment by j3ll3yfi5h** [comment]
@fabpot Not sure, but could be.

Is it not possible to merge something in 3.4 anymore? Because it took me quite a while to find this bug and maybe it is for someone else helpful...

**Comment by stof** [comment]
We don't merge things anymore in 3.4 as it is not maintained anymore (even security fixes are not done in 3.4 anymore).

**Comment by xabbuh** [comment]
The code basically looks the same on the `4.4` branch.

@j3ll3yfi5h Can you rebase your changes on the `4.4` branch and change the target branch here accordingly?

File: src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig [hunk 1]
```diff
@@ -379,18 +379,21 @@
...
 {%- block widget_attributes -%}
+    {{- " " -}}
     id="{{ id }}" name="{{ full_name }}"
     {%- if disabled %} disabled="disabled"{% endif -%}
     {%- if required %} required="required"{% endif -%}
     {{ block('attributes') }}
 {%- endblock widget_attributes -%}
 
 {%- block widget_container_attributes -%}
+    {{- " " -}}
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig:390]: shouldn't this be moved inside the next "if"?

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
For me this change look strange as the space is already there where the attributes are used eg. `<div {{ block('widg...` which I think the correct place where the space should be handled and not inside the attribute blocks. In which case/component the space is not there?

File: src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig (hunk 1 above)
Comment (j3ll3yfi5h) [reply src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig:390]: I don't think so.

The output of this `{{ widget_attributes }}` ended on my example with like `<divid ... >`

**Comment by nicolas-grekas** [comment]
@j3ll3yfi5h WDYT of @alexander-schranz's comment? It makes sense to me.

**Comment by j3ll3yfi5h** [comment]
> For me this change look strange as the space is already there where the attributes are used eg. `<div {{ block('widg...` which I think the correct place where the space should be handled and not inside the attribute blocks. In which case/component the space is not there?

@nicolas-grekas 

I came across this bug like so:
```
{%- block textarea_widget -%}
    <textarea {{ block('widget_attributes') }}>{{ value }}</textarea>
{%- endblock textarea_widget -%}
```

Outputs this:
`<textareaid="form_nachricht" name="form[nachricht]" required="required" class="form-control" cols="43" rows="4">`

Other example outputs, but less bad, because it's between two attributes:

1. `<input type="radio"id="form_anrede_0" name="form[anrede]" required="required" value="Herr" />`
2. `<input type="text"id="form_email" name="form[email]" required="required" class="form-control"/>`
3. `<button type="submit"id="form_submit" name="form[submit]" disabled="disabled" value="kontakt-anfrage" class="form_submit">Anfrage absenden</button>`

For me it looks like empty spaces (also like the 4 spacings/1 tab) between `{%- block XYZ -%}` and `{%- endblock XYZ -%}` are trimmed.

Because I can only prove the bug in **EDIT:** version ~~4.4~~ **3.4** I'm not sure if something changed in newer versions. Maybe also the twig parser changed? Does it output the space in newer versions?

**Comment by alexander-schranz** [comment]
@j3ll3yfi5h can you try to create a reproducable. In the `form_div_layout.html.twig` I can not reproduce your issue with the textarea: Example repo using Symfony 4.4: https://github.com/alexander-schranz/symfony-4-form-example

Textarea and buttons looks with attributes as it should:

Bildschirmfoto 2022-09-29 um 00 43 12

Maybe you have an issue in a custom theme?

**Comment by j3ll3yfi5h** [comment]
@alexander-schranz Thanks for checking and providing an example repo. Couldn't reproduce either.

A lot changed since 3.4, I assume that it's already fixed.

Thanks for your patience!

**Pull request closed without merging**